### PR TITLE
Status Icons

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Sorry, sammy sonic rider.

Adds an implementation for status icons, little icons that display on characters to inform people of various stuff about them.
Usages range from antag identifiers to sechud to medhud to whatever the hell you could envision.

This PR includes the basis for a zombie icon but I'm waiting on Electro's faction system before I actually implement it.
For now, this is just a code-side implementation.

Courtesy to RamZ for the zombie icon.

**Media**

image
sprite size-based cutoffs (it stops rendering past a certain point) (debug and using incorrect icon ordering)

image
multiple icons

image
working on non-standard sprite sizes (offset)

image
zombies

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

no cl no fun

### CODE REVIEW

**Comment by Cheackraze** [comment]
Could you make the system disable-able by cvar?

**Comment by EmoGarbage404** [comment]
> Could you make the system disable-able by cvar?

sure.

**Comment by Cheackraze** [comment]
> > Could you make the system disable-able by cvar?
> 
> sure.

Thanks! A lot of people find the icons visually distracting and powergamey so im certain that they will eventually want a way to disable this easily.

**Comment by Cheackraze** [comment]
Probably both. Clients for individual player preferences and server for balance reasons.

**Comment by AJCM-git** [comment]
> should the option be for individual clients or for servers as a whole? Maybe have both?

Sounds a little pointless for individual clients as an option as that would just make the game more difficult for them individually, would be better off as an trait in that case

**Comment by EmoGarbage404** [comment]
implemented 2 cvars:
- local_status_icons_enabled: clientside cvar for toggling status icons. could be a menu option in the future or something
- global_status_icons_enabled: server configuration for whether or not status icons should render.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Shared/StatusIcon/Components/StatusIconComponent.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+[RegisterComponent, NetworkedComponent]
+public sealed class StatusIconComponent : Component
+{
+    /// <summary>
+    /// Optional bounds for where the icons are laid out.
+    /// If null, the sprite bounds will be used.
+    /// </summary>
+    [DataField("bounds")]
+    public Box2? Bounds;
+}
```
[the remark this answers is not in the document]
Comment (EmoGarbage404) [reply Content.Shared/StatusIcon/Components/StatusIconComponent.cs:18]: can't do access because shared comp and clientside-only system

**Comment by EmoGarbage404** [comment]
added access and addressed all review. Should be gtg.

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**